docs/source/format/Glossary.rst

```diff
@@ -0,0 +1,207 @@
+       known length where all values have the same type.  Consists of
+       zero or more :term:`arrays <array>`, the "chunks".
+
+       Chunked arrays are discontiguous in the sense that iterating
+       the values of a chunked array may require iterating through
+       different buffers for different indices.
+
+       Not part of the columnar format; this term is specific to
+       certain language implementations of Arrow (primarily C++ and
+       its bindings).
+
+       .. seealso:: :term:`record batch`, :term:`table`
+
+   complex type
+   nested type
+       A :term:`data type` whose structure depends on one or more
+       other :term:`child data types <child type>`. For instance,
+       ``List`` is a nested type that has one child.
+
+       Two nested types are equal if and only if their child types are
+       also equal.
+
+   data type
+   type
+       A type that a value can take, such as ``Int8`` or
+       ``List[Utf8]``. The type of an array determines how its values
+       are laid out in memory according to :doc:`./Columnar`.
+
+       .. seealso:: :term:`logical type`, :term:`nested type`,
+                    :term:`primitive type`
+
+   dictionary
+       An array of values that accompany a :term:`dictionary-encoded
+       <dictionary-encoding>` array.
+
+   dictionary-encoding
+       An array that stores its values as indices into a
+       :term:`dictionary` array instead of storing the values
+       directly.
+
+       .. seealso:: :ref:`dictionary-encoded-layout`
+
+   extension type
+       A user-defined :term:`data type` that adds additional semantics
+       to an existing data type.  This allows implementations that do
+       not support a particular extension type to still handle the
+       underlying data type.
+
+       For example, a UUID can be represented as a 16-byte fixed-size
+       binary type.
+
+       .. seealso:: :ref:`format_metadata_extension_types`
+
+   field
+       A column in a :term:`record batch` or :term:`table`.  Consists
+       of a field name, a :term:`data type`, a flag indicating whether
+       the field is nullable or not, and optional key-value metadata.
+
+   IPC format
+       A specification for how to serialize Arrow data, so it can be
+       sent between processes/machines, or persisted on disk.
+
+   IPC file format
+   file format
+   random-access format
+       An extension of the :term:`IPC streaming format` that can be
+       used to serialize Arrow data to disk, then read it back with
+       random access to individual record batches.
+
+   IPC message
+   message
+       The IPC representation of a particular in-memory structure,
+       like a record batch or schema.
+
+   IPC streaming format
+   streaming format
+       A protocol for streaming Arrow data or for serializing data to
+       a file, consisting of a stream of :term:`IPC messages <IPC
+       message>`.
+
+   logical type
```

Comment:
I don't think we should have an entry for logical types as that notion doesn't exist at all (neither in the format, nor in C++).

Reply:
But we need to use _some_ term to distinguish, unless we would say something like that "other types with the same physical memory layout" can be found in schema.fbs

Reply:
There is something to distinguish at the _documentation_ level. Some types are listed in https://arrow.apache.org/docs/dev/format/Columnar.html#physical-memory-layout, others are referred to Schema.fbs. So that doesn't necessarily need to use the term "logical", but we need some phrasing about this in the documentation.

Reply:
As someone who's been confused every time I encountered the term "logical type" in the columnar format doc page (without any corresponding term in the actual spec or C++ code), I don't want to see it reiterated in the glossary.

Reply:
As a sidenote, I think it would still be beneficial to overhaul the columnar doc page to make it more approachable and less confusing, but it needs a skilled writer that is also acquainted with Arrow. Not easy to find? cc @amol- for ideas.